docs/website/blog/2023-12-13-dlt-aws-taktile-blog.md

```diff
@@ -0,0 +1,139 @@
+    pip install aws-sam-cli
+    ```
+    
+2. Define your resources in a `template.yml` file
+    
+    ```yaml
+    AWSTemplateFormatVersion: "2010-09-09"
+    Transform: AWS::Serverless-2016-10-31
+    
+    Resources:
+      ApiGateway:
+        Type: AWS::Serverless::Api
+        Properties:
+          Name: DLT Api Gateway
+          StageName: v1
+      DltFunction:
+        Type: AWS::Serverless::Function
+        Properties:
+          PackageType: Image
+          Timeout: 30 # default is 3 seconds, which is usually too little
+          MemorySize: 512 # default is 128mb, which is too little
+          Events:
+            HelloWorldApi:
+              Type: Api
+              Properties:
+                RestApiId: !Ref ApiGateway
+                Path: /collect
+                Method: POST
+          Environment:
+            Variables:
+              DLT_PROJECT_DIR: "/tmp" # the only writeable directory on a Lambda
+              DLT_DATA_DIR: "/tmp" # the only writeable directory on a Lambda
+              DLT_PIPELINE_DIR: "/tmp" # the only writeable directory on a Lambda
+              LOAD__WORKERS: 1 # AWS Lambda cannot do multiprocessing
+          Policies:
+            - Statement:
+                - Sid: AllowDLTSecretAccess
+                  Effect: Allow
+                  Action:
+                    - secretsmanager:GetSecretValue
+                  Resource: !Sub arn:aws:secretsmanager:${AWS::Region}:${AWS::AccountId}:secret:DLT_*
+        Metadata:
+          DockerTag: dlt-aws
+          DockerContext: .
+          Dockerfile: Dockerfile
+    Outputs:
+      ApiGateway:
+        Description: "API Gateway endpoint URL for Staging stage for Hello World function"
+        Value: !Sub "https://${ApiGateway}.execute-api.${AWS::Region}.amazonaws.com/v1/collect/"
+    ```
+    
+3. Build a deployment package
+    
+    ```bash
+    sam build
+    ```
+    
+4. Test your setup locally
+    
+    ```bash
+    sam local start-api
+    
+    # in a second terminal window
+    curl -X POST http://127.0.0.1:3000/collect -d '{"hello":"world"}'
+    ```
+    
+5. Deploy your resources to AWS
+    
+    ```bash
+    sam deploy --stack-name=<your-stack-name> --resolve-image-repos --resolve-s3 --capabilities CAPABILITY_IAM
+    ```
+    
+
+### Caveats to be aware of when setting up dlt on AWS Lambda:
+
+No worries, all caveats described below are already being taken care of in the sample repo: https://github.com/codingcyclist/dlt-aws-lambda. I still recommend you read through them to be aware of what’s going on.
+
+1. **Multiprocessing:** While AWS Lambda functions scale very flexibly by spawning multiple runners in parallel, each Lambda function only runs on a single process. By default, dlt will try to spawn multiple processes. To overwrite dlt’s default behavior, simply set the `LOAD__WORKERS` environment variable to `1`. Your data will still get loaded quickly as AWS spawns additional instances of your function when needed.
```

Comment:
We can take out this paragraph. Dlt now uses a different kind of Threadpool under the hood that auto-detects that Lambda has only 1 process to run on